test/aleph/resource_leak_detector.clj

```diff
@@ -0,0 +1,210 @@
+(ns aleph.resource-leak-detector
+  "Provides a Netty leak detector which is more reliable than the default implementation.
+
+  Its chance of detecting leaks is around 95%. See `with-leak-collection` for details.
+
+  Since it adds considerable runtime overhead, its main purpose is to be used in tests to assert
+  intentional presence of leaks and to check for unintended leaks. The main API entry points for
+  this are `with-leak-collection`, `with-expected-leaks` and `instrument-tests!`.
+
+  To enable it, pass `-Dio.netty.customResourceLeakDetector=aleph.resource_leak_detector` to the
+  JVM. You will also have to pass `-Dio.netty.leakDetection.level=PARANOID`.
+
+  Most reliable results are achieved by also passing the following options:
+  `-Dio.netty.leakDetection.targetRecords=1`
+  `-Dio.netty.allocator.type=unpooled`
+
+  All of the above is also provided by the `:leak-detection` Leiningen profile.
```

Comment:
Minor: Excellent docs, but I suggest reversing the order here. Mention the :leak-detection profile first, then go into what it's doing under the hood. (I assume 99% of the time we'll just want to add `with-profile +leak-detection`?)